### ISSUE TITLE
InterruptedException while creating index

### ISSUE THREAD

**Issue by ronlut** [neo4j/neo4j#10251]
- Neo4j version: 3.2.6
- Operating system: Ubuntu 16.04.1
- Expected behavior: Index creation
- Actual behavior: Index creation fails after less than 2 minutes

So, I'm trying to create a simple single-property index on the _value_ property of _:Node_ using:
`create index on :Node(value)`
but after less than 3 minutes i'm getting an InterruptedException and it fails.

Log:

> 2017-10-10 15:34:57.394+0000 INFO [o.n.k.i.a.i.IndexPopulationJob] Index population started: [:Node(value) [provider: {key=lucene, version=1.0}]]
2017-10-10 15:36:26.549+0000 INFO [o.n.k.i.a.i.IndexPopulationJob] Forcefully shutting down executor.
BatchingMultipleIndexPopulator{activeTasks=1, executor=java.util.concurrent.ThreadPoolExecutor@d7469ca[Running, pool size = 3, active threads = 1, queued tasks = 0, completed tasks = 2496], batchedUpdates = [8864 updates], queuedUpdates = 0}
2017-10-10 15:36:26.976+0000 ERROR [o.n.k.i.a.i.IndexPopulationJob] Failed to populate index: [:Node(value) [provider: {key=lucene, version=1.0}]] java.lang.InterruptedException
org.apache.lucene.util.ThreadInterruptedException: java.lang.InterruptedException
        at org.apache.lucene.index.IndexWriter.doWait(IndexWriter.java:4305)
        at org.apache.lucene.index.IndexWriter.abortMerges(IndexWriter.java:2272)
        at org.apache.lucene.index.IndexWriter.rollbackInternalNoCommit(IndexWriter.java:2053)
        at org.apache.lucene.index.IndexWriter.rollbackInternal(IndexWriter.java:2041)
        at org.apache.lucene.index.IndexWriter.tragicEvent(IndexWriter.java:4553)
        at org.apache.lucene.index.IndexWriter.updateDocuments(IndexWriter.java:1330)
        at org.apache.lucene.index.IndexWriter.addDocuments(IndexWriter.java:1297)
        at org.neo4j.kernel.api.impl.schema.writer.PartitionedIndexWriter.addDocuments(PartitionedIndexWriter.java:64)
        at org.neo4j.kernel.api.impl.schema.populator.LuceneIndexPopulator.add(LuceneIndexPopulator.java:68)
        at org.neo4j.kernel.impl.api.index.BatchingMultipleIndexPopulator.lambda$flush$3(BatchingMultipleIndexPopulator.java:237)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
        at org.neo4j.helpers.NamedThreadFactory$2.run(NamedThreadFactory.java:109)
Caused by: java.lang.InterruptedException
        at java.lang.Object.wait(Native Method)
        at org.apache.lucene.index.IndexWriter.doWait(IndexWriter.java:4303)
        ... 13 more

I can't understand why it's happening, as I was able to create a similar index on the _key_ propery of _:Node_

It seems like something is stopping it middle way, what can it be?

**Comment by TomGroves** [comment]
Possibly related?  [Attempting to create a uniqueness constraint causes Neo4J to hang.](https://github.com/neo4j/neo4j/issues/10276)
I don't even get anything nice like an exception, it just hangs. And it's not because it's busy - it's a brand new, empty graph.

**Comment by utkuc** [comment]
This issue was moved to utkuc/neo4j#2

**Comment by burqen** [comment]
@ronlut This indeed sound very strange. I have some questions:
- Have you been able to reproduce it? If so, how?
- Do you have the complete logs from the event? The most interesting one would be `debug.log`. The stacktrace that you provided only tells me _that_ it happened, but I can't see _how_.
- Are you running single instance or in a cluster? If you are running in a cluster, logs from other machines at the time of the event is very interesting.
- What other activities was there at the time of the event?
- How big is your store?

@TomGroves Could be related, yes. However, my gut feeling says no. Not jumping to any conclusions.

**Comment by FoysalOsmany** [comment]
Facing similar issue when trying to create index on Big data set. I am trying to create an index where number of nodes in 18 Million. But the actual problem is I was able to create index for the same node with other properties. So some property is able to create index others didn't.

- Neo4j version: 3.2.2
- Operating system: Ubuntu 16
- AWS EC2 m4:2xl

![image](https://user-images.githubusercontent.com/1332990/33179442-a7212ad8-d093-11e7-8c58-afb9bfec7199.png)

`2017-11-23 14:51:35.260+0000 INFO [o.n.k.i.a.i.IndexPopulationJob] Index population started: [:Business(domain) [provider: {key=lucene, version=1.0}]]
2017-11-23 14:52:35.240+0000 INFO [o.n.k.i.a.i.IndexPopulationJob] Forcefully shutting down executor.
BatchingMultipleIndexPopulator{activeTasks=2, executor=java.util.concurrent.ThreadPoolExecutor@dbec966[Running, pool size = 7, active threads = 2, queued tasks = 0, completed tasks = 1306], batchedUpdates = [177 updates], queuedUpdates = 0}
2017-11-23 14:52:35.863+0000 ERROR [o.n.k.i.a.i.IndexPopulationJob] Failed to populate index: [:Business(domain) [provider: {key=lucene, version=1.0}]] java.lang.InterruptedException
org.apache.lucene.util.ThreadInterruptedException: java.lang.InterruptedException
	at org.apache.lucene.index.IndexWriter.doWait(IndexWriter.java:4305)
	at org.apache.lucene.index.IndexWriter.abortMerges(IndexWriter.java:2272)
	at org.apache.lucene.index.IndexWriter.rollbackInternalNoCommit(IndexWriter.java:2053)
	at org.apache.lucene.index.IndexWriter.rollbackInternal(IndexWriter.java:2041)
	at org.apache.lucene.index.IndexWriter.tragicEvent(IndexWriter.java:4553)
	at org.apache.lucene.index.IndexWriter.updateDocuments(IndexWriter.java:1330)
	at org.apache.lucene.index.IndexWriter.addDocuments(IndexWriter.java:1297)
	at org.neo4j.kernel.api.impl.schema.writer.PartitionedIndexWriter.addDocuments(PartitionedIndexWriter.java:64)
	at org.neo4j.kernel.api.impl.schema.populator.LuceneIndexPopulator.add(LuceneIndexPopulator.java:68)
	at org.neo4j.kernel.impl.api.index.BatchingMultipleIndexPopulator.lambda$flush$3(BatchingMultipleIndexPopulator.java:237)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
	at org.neo4j.helpers.NamedThreadFactory$2.run(NamedThreadFactory.java:109)
Caused by: java.lang.InterruptedException
	at java.lang.Object.wait(Native Method)
	at org.apache.lucene.index.IndexWriter.doWait(IndexWriter.java:4303)
	... 13 more`

**Comment by burqen** [comment]
@FoysalOsmany Thank you for reporting. I will ask the same thing from you to be able to continue investigation:

- Have you been able to reproduce it? If so, how?
- Do you have the complete logs from the event? The most interesting one would be debug.log. The stacktrace that you provided only tells me that it happened, but I can't see how.
- Are you running single instance or in a cluster? If you are running in a cluster, logs from other machines at the time of the event is very interesting.
- What other activities was there at the time of the event?
- How big is your store?

**Comment by FoysalOsmany** [comment]
@burqen Still having the issue.

Have you been able to reproduce it? If so, how?

> This issue is easily reproducible each time I trying to create index for domain in Business node I am getting this issue.

 Do you have the complete logs from the event? The most interesting one would be debug.log. The stacktrace that you provided only tells me that it happened, but I can't see how.

> debug.log is attached [debug.log](https://github.com/neo4j/neo4j/files/1502853/debug.log)

Are you running single instance or in a cluster?

> Yes

What other activities was there at the time of the event?

> No other activities was there

How big is your store?

> Currently its 30gb containing 25 million nodes and relations. The issue facing for create index on Business node containing 10million.

**Comment by burqen** [comment]
@FoysalOsmany Thank you!

I should have been more specific about the possibility to reproduce. I meant to ask, **do you have a step by step process to reproduce the issue from an empty store so that I can continue the investigation locally on my end?**

I don't understand your answer about clustering. **Are you running in single instance mode or are you running in a cluster?**

Thanks for the debug log. It seems like this is not the complete log though. Is that correct?
From what I can see however is that on startup the index is already in a failed state
```
2017-11-25 06:20:00.123+0000 INFO [o.n.k.i.a.i.IndexingService] IndexingService.start: index 7 on :Business(domain) is FAILED
```
due to the record not being in use
```
2017-11-25 06:22:28.107+0000 INFO [o.n.k.i.a.i.FailedIndexProxy] FailedIndexProxy#drop index on :label[7](property[26]) [provider: {key=lucene+native, version=1.0}] dropped due to:
native: org.neo4j.kernel.impl.store.InvalidRecordException: DynamicRecord[2713842,used=false,(0),type=-1,data=byte[],start=true,next=-1] not in use
```
and it needs to be dropped
```
2017-11-25 06:22:23.655+0000 ERROR [o.n.b.v.r.ErrorReporter] Client triggered an unexpected error [IndexCorruptionDetected]: Index `Index( GENERAL, :Business(domain) )` has failed. Drop and recreate it to get it back online., reference 5081badd-74d5-4bd5-8921-4e753af6df99.
```
From the log it looks like it is dropped. **Can you confirm that you explicitly drop the index manually?**
```
2017-11-25 06:22:28.107+0000 INFO [o.n.k.i.a.i.FailedIndexProxy] FailedIndexProxy#drop index on :label[7](property[26]) [provider: {key=lucene+native, version=1.0}] dropped due to:
```
In between `06:22:33` and `06:23:40` the thread writing to Lucene is interrupted and cause the population job to be forcefully shutdown.
```
2017-11-25 06:22:33.957+0000 INFO [o.n.k.i.a.i.IndexPopulationJob] Index population started: [:Business(domain) [provider: {key=lucene+native, version=1.0}]]
2017-11-25 06:22:39.927+0000 WARN [o.n.k.i.c.MonitorGc] GC Monitor: Application threads blocked for 280ms.
2017-11-25 06:23:40.096+0000 INFO [o.n.k.i.a.i.IndexPopulationJob] Forcefully shutting down executor.
BatchingMultipleIndexPopulator{activeTasks=2, executor=java.util.concurrent.ThreadPoolExecutor@4485d4f6[Running, pool size = 7, active threads = 2, queued tasks = 0, completed tasks = 1306], batchedUpdates = [177 updates], queuedUpdates = 0}
2017-11-25 06:23:40.332+0000 ERROR [o.n.k.i.a.i.IndexPopulationJob] Failed to populate index: [:Business(domain) [provider: {key=lucene+native, version=1.0}]] java.lang.InterruptedException
```
Just to make sure. **Did you manually interrupt any process during this period?**

**Comment by burqen** [comment]
Closing this issue due to lack of answer. Please feel free to open this again, should you feel the need to.
